### PR TITLE
Add display filter to definition expression sample

### PR DESCRIPTION
**Pull request by Rachael-E**
This PR enhances and updates the existing Feature layer definition expression sample:

- it includes the functionality to use display filters on the feature layer in addition to the definition expression functionality
- it renames the sample to "Filter by definition expression or display filter"
- new UI to handle the switching between the different display modes, along with a reset button and a counted features label
- the map view is navigation is disabled so that the focus for the user is on the data and how it changes depending on what filter is selected
- replaces image of sample and updates metadatajson and readme
- since this sample is being renamed in this PR, the redirect in the metadata has been updated

@jenmerritt could you take a look please?

tagging @jasongao1230 and @kathleen1 for upcoming GUI change alert.

### CODE REVIEW

File: feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2021 Esri.
```
Comment (jenmerritt) [inline feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java]:
```suggestion
 * Copyright 2022 Esri.
```

File: feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+public class FilterByDefinitionExpressionOrDisplayFilterSample extends Application {
+  
+  private Button applyExpressionButton;
+  private Button applyFilterButton;
+  private Button resetButton;
+  private Label featureCountLabel;
+
+  private MapView mapView;
+  private FeatureLayer featureLayer;
```
Comment (jenmerritt) [inline feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java]: Could this be a local variable?

File: feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+  private final static String FEATURE_SERVICE_URL =
```
Comment (jenmerritt) [inline feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java]:
```suggestion
  private static final String FEATURE_SERVICE_URL =
```
Usual order for Java

**Review by jenmerritt — APPROVED** [review]
LGTM @Rachael-E , couple of minor tidy ups suggested

File: feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java (hunk 2 above)
Comment (Rachael-E) [reply feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java]: oh good catch: it needs the garbage collection comment next to it!

**Pull request merged**